src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/Powertrain.java

```diff
@@ -0,0 +1,126 @@
+package hu.oe.nik.szfmv.automatedcar.systemcomponents;
+
+import com.github.pyknic.vector.Vec2f;
+import hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.VirtualFunctionBus;
+
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.List;
+
+public class Powertrain extends SystemComponent implements IPowertrain {
+    private static final int RPM_CONSTANT = 60;
+    private static final int BRAKE_CONSTANT = 60;
+    private static final int RESIST_FORCE_CONSTANT = 100;
+    private static final int CAR_MASS = 1000;
+    private static final int WHEELBASE = 130;
+    private static final int CAR_WIDTH = 90;
+    private static final Vec2f NULL_VECTOR = Vec2f.constant(0, 0);
+    private static final Vec2f FORWARD_VECTOR = Vec2f.constant(0, 1);
+    private static final Vec2f BACKWARD_VECTOR = Vec2f.constant(0, -1);
+    private static final List<Double> GEAR_RATIOS = Arrays.asList(2.66, 1.78, 1.3, 1.0, 0.74, 2.9);
+    private static final List<AbstractMap.SimpleEntry> INSIDE_GEAR_SHIFT_LIMITS = Arrays.asList(
+            new AbstractMap.SimpleEntry(0, 6000),
+            new AbstractMap.SimpleEntry(3385, 6000),
+            new AbstractMap.SimpleEntry(4010, 6000),
+            new AbstractMap.SimpleEntry(4356, 6000),
+            new AbstractMap.SimpleEntry(4224, Integer.MAX_VALUE)
+    );
```

Comment:
Ez miért AbstractMap? Elvileg nem is lehet példányosítani, mert abstract.
A HashMap oké, de az meg nem garantál rendezettséget.
Egyszerűen nagyon erőltetett az egész konstrukció.

Azért kezdtem komolyabban vizsgálni, mert a getCurrentInsideGearShift-ben a castolásnál nekem konkrétan elszáll a kód. Nem tudom, hogy mi más nálam mint a CI környezetben, de valami nem kóser.

Nem megoldás, hogy (nem valid java szintaxis) LOWER_LIMIT = [0, 3385, 4010, 4356, 4224], UPPER_LIMIT = [6000,6000,6000,Integer.MAX_VALUE] és a LOWER_LIMIT[gear_position] módon eléritek?